lib/mcp/server.rb

```diff
@@ -47,13 +48,19 @@ def initialize(
     )
       @name = name
       @version = version
+      @instructions = instructions
       @tools = tools.to_h { |t| [t.name_value, t] }
       @prompts = prompts.to_h { |p| [p.name_value, p] }
       @resources = resources
       @resource_templates = resource_templates
       @resource_index = index_resources_by_uri(resources)
       @server_context = server_context
       @configuration = MCP.configuration.merge(configuration)
+      if @configuration.protocol_version == "2024-11-05" && @instructions
```

Comment:
Stylistic subjective nitpick:

```suggestion
      @configuration = MCP.configuration.merge(configuration)
      
      if @configuration.protocol_version == "2024-11-05" && @instructions
```